### ISSUE TITLE
Linking straight to anchors in the documentation

### ISSUE THREAD

**Issue by JamesThorpe** [knockout/knockout#1763]
Quite often when answering questions on Stack Overflow, it's useful to link to a specific area of the documentation.  Most of the areas have anchors specific to them, eg: http://knockoutjs.com/documentation/click-binding.html#note-3-allowing-the-default-click-action

However there's no easy way to link straight to this, other than either typing it by hand and hoping you get it right, or viewing the source and grabbing the ID out.  Could the documentation be updated to include a small link to each of these headings?

**Comment by SteveSanderson** [comment]
If you can suggest a good way of doing that without handcoding the links everywhere in the docs, I'd be entirely open to it! Do you want to submit a pull request?

**Comment by JamesThorpe** [comment]
The documentation source aren't something I've looked at (yet) - are they run through a build process of some sort?  But yes, I'll take a look if I can!

**Comment by mbest** [comment]
It would be gh-pages.

**Comment by alvarotrigo** [comment]
Where is the documentation placed?  Where is `gh-pages`

**Comment by rniemeyer** [comment]
@alvarotrigo - It is the `gh-pages` branch of the knockout repo, which you can find here: https://github.com/knockout/knockout/tree/gh-pages .  Thanks!

**Comment by JamesThorpe** [comment]
I completely forgot about this... I believe I got as far as seeing that Kramdown could be made to do this, but by supplying a custom link rendered that overrides the built in one.  The issue would be whether or not Jekyll could be made to invoke Kramdown to use it during the build process without modifying Jekyll itself.

**Comment by alvarotrigo** [comment]
@JamesThorpe I'm not familiarized with Jekyll. But I believe a very easy solution is to use javascript / jQuery to do it.

Demo: http://jsfiddle.net/zetr1c68/

```javascript
$('h1, h2, h3').each(function(){
    $(this).append('<a href="#'+ this.id +'" class="anchorlink"><span class="octicon octicon-link">¶</span></a>');    
});
```

```css
h1:hover .anchorlink,
h2:hover .anchorlink,
h3:hover .anchorlink{
    display: inline;
}
a.anchorlink{
    display: none;
    color: #999;
    text-decoration:none;
    margin:0 0 0 5px;
}
a.anchorlink:hover{
    color: #3469BD;
}
```
I used jQuery in the example as I saw knockout website is already using it.

**Comment by JamesThorpe** [comment]
@alvarotrigo Yes that would work - I was approaching it from the other angle :)  Guess it's down to what the guys here would find acceptable.  I think a client-side based approach like yours would be perfectly fine for this.

**Comment by brianmhunt** [comment]
This is definitely on the TODO. :) Thanks for the insight and comments, all.
